packages/worker/src/token/token.service.ts

```diff
@@ -97,17 +99,27 @@ export class TokenService {
         tokenAddress: contractAddress.address,
       });
 
-      await this.tokenRepository.upsert({
-        ...erc20Token,
-        blockNumber: contractAddress.blockNumber,
-        transactionHash: contractAddress.transactionHash,
-        l2Address: contractAddress.address,
-        logIndex: contractAddress.logIndex,
-        // add L1 address for ETH token
-        ...(contractAddress.address.toLowerCase() === utils.L2_BASE_TOKEN_ADDRESS && {
-          l1Address: utils.ETH_ADDRESS,
-        }),
-      });
+      if (contractAddress.address.toLowerCase() === utils.L2_BASE_TOKEN_ADDRESS.toLowerCase()) {
+        await this.tokenRepository.upsert({
+          blockNumber: contractAddress.blockNumber,
+          transactionHash: contractAddress.transactionHash,
+          logIndex: contractAddress.logIndex,
+          l2Address: utils.L2_BASE_TOKEN_ADDRESS,
+          l1Address: this.configService.get<string>("tokens.baseToken.l1Address") ?? utils.ETH_ADDRESS,
```

Comment:
No need for default values as the config `tokens.baseToken.*` already has defaults